src/common/buffer.cc

```diff
@@ -135,24 +135,33 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
   buffer::error_code::error_code(int error) :
     buffer::malformed_input(cpp_strerror(error).c_str()), code(error) {}
 
+#define CRC_CACHE_THRESHOLD (2 * CEPH_PAGE_SIZE)
+
   class buffer::raw {
   public:
     char *data;
     unsigned len;
     atomic_t nref;
 
-    mutable RWLock crc_lock;
-    map<pair<size_t, size_t>, pair<uint32_t, uint32_t> > crc_map;
+    mutable RWLock* crc_lock;
+    map<pair<size_t, size_t>, pair<uint32_t, uint32_t> >* crc_map;
+
+    raw(unsigned l) : raw(NULL, l) { }
 
-    raw(unsigned l)
-      : data(NULL), len(l), nref(0),
-	crc_lock("buffer::raw::crc_lock", false)
-    { }
     raw(char *c, unsigned l)
-      : data(c), len(l), nref(0),
-	crc_lock("buffer::raw::crc_lock", false)
-    { }
-    virtual ~raw() {}
+      : data(c), len(l), nref(0), crc_lock(NULL), crc_map(NULL)
+    {
+		if(l > CRC_CACHE_THRESHOLD) {
+			crc_lock = new RWLock("buffer::raw::crc_lock", false);
+			crc_map = new map<pair<size_t, size_t>, pair<uint32_t, uint32_t>>;
+		}
+	}
+    virtual ~raw() {
+		if(crc_lock)
+			delete crc_lock;
+		if(crc_map)
+			delete crc_map;
+	}
```

Comment:
Lots of style and whitespace issues.  Please use an editor that respects the modeline at the top of the file (it works with vim and emacs) and see the CodingStyle doc